jenkins_jobs/upload_build_artifacts/script.sh

```diff
@@ -1,34 +1,30 @@
-BUILD_TIMESTAMP=$(cat ./BUILD_TIMESTAMP)
-BASE_URL="http://${UPLOAD_SERVER_HOST_NAME}${UPLOAD_SERVER_BUILDS_DIR}/${BUILD_TIMESTAMP}"
+BUILD_TIMESTAMP=$(cat BUILD_TIMESTAMP)
+BUILD_DIR_PATH="${UPLOAD_SERVER_BUILDS_DIR}/${BUILD_TIMESTAMP}"
+BUILD_DIR_HTTP_URL="http://${UPLOAD_SERVER_HOST_NAME}${BUILD_DIR_PATH}"
+BUILD_DIR_RSYNC_URL="${UPLOAD_SERVER_USER_NAME}@${UPLOAD_SERVER_HOST_NAME}:${BUILD_DIR_PATH}"
 
 if [ -f SUCCESS ]; then
-    REPOSITORY_FILE_URL="${BASE_URL}/hostos.repo"
+    REPOSITORY_FILE_URL="${BUILD_DIR_HTTP_URL}/hostos.repo"
     BUILD_STATUS='PASS'
 else
     REPOSITORY_FILE_URL='null'
     BUILD_STATUS='FAIL'
 fi
 echo  "{'REPO': '$REPOSITORY_FILE_URL', 'BUILD_TIMESTAMP': '${BUILD_TIMESTAMP}', 'BUILD_LOG': '$BUILD_URL', 'BUILD_ID': $BUILD_JOB_NUMBER, 'BUILD_STATUS': '$BUILD_STATUS'}" > STATUS
 
-rsync -e "ssh -i $HOME/.ssh/${UPLOAD_SERVER_USER_NAME}_id_rsa" \
-      --verbose --compress --stats --times --chmod=a+rwx,g+rwx,o- \
-      "STATUS" \
-      "${UPLOAD_SERVER_USER_NAME}@${UPLOAD_SERVER_HOST_NAME}:${UPLOAD_SERVER_BUILDS_DIR}/${BUILD_TIMESTAMP}/"
+alias rsync_upload="rsync -e \
+    'ssh -i ${HOME}/.ssh/${UPLOAD_SERVER_USER_NAME}_id_rsa' \
+    --verbose --compress --stats --times --chmod=a+rwx,g+wx,o-"
 
-rsync -e "ssh -i $HOME/.ssh/${UPLOAD_SERVER_USER_NAME}_id_rsa" \
-      --verbose --compress --recursive --stats --times --chmod=a+rwx,g+rwx,o- \
-      "repository" \
-      "${UPLOAD_SERVER_USER_NAME}@${UPLOAD_SERVER_HOST_NAME}:${UPLOAD_SERVER_BUILDS_DIR}/${BUILD_TIMESTAMP}/"
+rsync_upload STATUS ${BUILD_DIR_RSYNC_URL}/
+rsync_upload --recursive repository ${BUILD_DIR_RSYNC_URL}/
```

Comment:
It's needed in the first command to create the build directory. I am removing it and making the directory creation explicit to avoid confusion.